smart_open/tests/test_smart_open.py

```diff
@@ -259,7 +259,34 @@ def test_http_bz2(self):
                       body=compressed_data, stream=True)
         smart_open_object = smart_open.smart_open("http://127.0.0.1/data.bz2")
 
-        # decompress the gzip and get the same md5 hash
+        # decompress the bzip2 and get the same md5 hash
+        self.assertEqual(smart_open_object.read(), test_string)
+
+    @responses.activate
+    def test_http_xz(self):
+        """Can open xz via http?"""
+        test_string = b'Hello World Compressed.'
+        #
```

Comment:
Yes, I agree with the text in this TODO. We should avoid unnecessary file IO (create temporary file, write to file, read from file, delete file) in unit tests, especially when the alternative is cleaner and simpler (encode to a io.BytesIO directly).

Reply:
This has been copied from `test_http_bz`. I understand that you wish to refactor the tests, but the goal of this PR is different. I will submit a refactoring PR separately, ok?